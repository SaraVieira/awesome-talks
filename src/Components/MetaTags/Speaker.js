import React from 'react'
import { Helmet } from 'react-helmet'

export default ({ name, photo }) => (
    <Helmet>
        <title>Awesome Talks - {name}</title>
        <meta name="twitter:title" content={`Awesome Talks - ${name}`} />
        <meta name="twitter:image" content={photo.url} />
        <meta name="twitter:image:alt" content={name} />
        <meta name="description" content={`Amazing Tech Talks by ${name}`} />
        <meta
            name="twitter:description"
            content={`Amazing Tech Talks by ${name}`}
        />
    </Helmet>
)
