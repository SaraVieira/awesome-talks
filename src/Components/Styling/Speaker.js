import styled from 'styled-components'

export const Figure = styled.figure`
    position: relative;
    float: left;
    overflow: hidden;
    margin: 10px 1%;
    min-width: 220px;
    max-width: 220px;
    max-height: 220px;
    min-height: 220px;
    width: 100%;
    background: ${props => props.theme.main};
    color: ${props => props.theme.primary};
    text-align: center;
    box-shadow: 0 0 5px rgba(0, 0, 0, 0.15);
    background: ${props => props.theme.blue};
    &:hover img {
        opacity: 1;
        transform: scale(1.1);
    }

    &:hover h2 {
        transform: skew(-10deg) rotate(-10deg) translate(-150%, -50%);
    }

    &:hover:before {
        transform: rotate(110deg) translateY(-150%);
    }
    &::before {
        height: 120%;
        width: 130%;
        top: 0;
        left: 0;
        content: '';
        background: ${props => props.theme.primary};
        position: absolute;
        transition: all 0.3s ease-in-out;
        transform: rotate(110deg) translateY(-33%) translateX(-7px);
    }

    * {
        box-sizing: border-box;
        transition: all 0.4s ease-in-out;
    }

    a {
        left: 0;
        right: 0;
        top: 0;
        bottom: 0;
        position: absolute;
        z-index: 1;
        padding: 0;
        border: none;
    }
`

export const Img = styled.img`
    max-width: 100%;
    position: relative;
    opacity: 0.4;
`

export const Caption = styled.figcaption`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
`

export const Name = styled.h2`
    position: absolute;
    left: 20px;
    right: 20px;
    display: inline-block;
    background: ${props => props.theme.main};
    transform: skew(-10deg) rotate(-10deg) translate(0, -50%);
    padding: 12px 5px;
    margin: 0;
    bottom: 0;
    text-transform: uppercase;
    font-weight: 400;
    span {
        font-weight: 800;
        display: block;
    }
`
