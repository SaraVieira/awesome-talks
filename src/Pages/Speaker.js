import React from 'react'
import Header from './../Components/Header'
import { Col, Row, Grid } from 'react-styled-flexboxgrid'
import Flex from 'styled-flex-component'
import styled from 'styled-components'
import remcalc from 'remcalc'
import { Helmet } from 'react-helmet'

import Query from './../Components/Query'
import Video from './../Components/Video'
import SPEAKER_VIDEOS from '../Queries/SPEAKER_VIDEOS'
import TwitterIcon from '../assets/twitter.svg'
import humanize, { urlify } from '../Utils/strings'
import Nav from './../Components/Nav'
import CookieBanner from './../Components/CookieBanner'
import Error404 from './../Components/Errors/Error404'

const Wrapper = styled(Row)`
    margin-bottom: ${remcalc(30)};

    * {
        box-sizing: border-box;
    }

    @media (max-width: ${remcalc(768)}) {
        justify-content: center;
    }
`

const Img = styled.img`
    margin-right: ${remcalc(20)};
    box-shadow: ${props => props.theme.shadow};
    width: ${remcalc(200)};
    min-width: ${remcalc(200)};

    @media (max-width: ${remcalc(768)}) {
        margin: auto;
    }
`

const Desc = styled(Flex)`
    @media (max-width: ${remcalc(768)}) {
        flex-direction: column;

        h1 {
            margin-top: 10px;
            margin-bottom: 0;
        }
    }
`

const Section = styled.div`
    width: 100%;

    p {
        word-break: break-word;
    }

    @media (max-width: ${remcalc(768)}) {
        max-width: 80%;
        margin: auto;

        p {
            word-break: break-word;
            text-align: center;
        }
    }
`

const SpeakerInfo = ({ photo, name, bio, twitter }) => (
    <Wrapper>
        <Helmet>
            <title>Awesome Talks - {name}</title>
            <meta name="twitter:title" content={`Awesome Talks - ${name}`} />
            <meta name="twitter:image" content={photo.url} />
            <meta name="twitter:image:alt" content={name} />
            <meta
                name="description"
                content={`Amazing Tech Talks by ${name}`}
            />
            <meta
                name="twitter:description"
                content={`Amazing Tech Talks by ${name}`}
            />
        </Helmet>
        <Desc>
            {photo ? (
                <Img src={photo.url} alt={name} height="200" width="200" />
            ) : null}
            <Section>
                <Header title={humanize(name)} noSearch small />
                <p dangerouslySetInnerHTML={{ __html: urlify(bio) }} />{' '}
                <Flex justifyBetween alignCenter>
                    {twitter ? (
                        <a
                            className="no-hover"
                            target="_blank"
                            rel="noopener noreferrer"
                            href={`https://twitter.com/${twitter}`}
                        >
                            <img src={TwitterIcon} alt="Twitter" width="24" />
                        </a>
                    ) : null}
                    <a
                        className="no-hover"
                        target="_blank"
                        rel="noopener noreferrer"
                        href={`https://github.com/SaraVieira/awesome-talks/issues/new?title=Wrong%20Speaker%20info%20for%20${name}&body=Hey!%20The%20info%20on%20${name}%20is%20wrong,%20what%20is%20wrong%20is`}
                    >
                        Wrong Info?
                    </a>
                </Flex>
            </Section>
        </Desc>
    </Wrapper>
)

export default ({
    match: {
        params: { speaker }
    }
}) => (
    <Grid>
        <div role="banner">
            <Nav />
        </div>
        <main>
            <Row>
                <Col xs={12}>
                    <Query
                        query={SPEAKER_VIDEOS}
                        variables={{ name: humanize(speaker) }}
                    >
                        {({ data: { allSpeakerses } }) => {
                            if (!allSpeakerses.length) {
                                return <Error404 />
                            }

                            return (
                                <Section>
                                    <SpeakerInfo {...allSpeakerses[0]} />
                                    <Row>
                                        <Header title="Talks" noSearch />
                                    </Row>
                                    <Row>
                                        {allSpeakerses.length &&
                                            allSpeakerses[0].videoses.length &&
                                            allSpeakerses[0].videoses.map(v => (
                                                <Video key={v.id} talk={v} />
                                            ))}
                                    </Row>
                                </Section>
                            )
                        }}
                    </Query>
                </Col>
            </Row>
        </main>
        <CookieBanner />
    </Grid>
)
