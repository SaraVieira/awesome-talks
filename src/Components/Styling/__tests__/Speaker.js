import React from 'react'
import Shallow from '../../../Tests/shallow'
import { Figure, Name } from '../Speaker'

test('Renders Figure', () => {
    const Wrapper = Shallow(<Figure />)

    expect(Wrapper).toMatchSnapshot()
})

test('Renders Name', () => {
    const Wrapper = Shallow(<Name />)

    expect(Wrapper).toHaveStyleRule('text-transform', 'uppercase')
    expect(Wrapper).toMatchSnapshot()
})
