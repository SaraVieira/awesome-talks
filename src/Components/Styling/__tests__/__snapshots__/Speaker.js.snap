// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Figure 1`] = `
.c0 {
  position: relative;
  float: left;
  overflow: hidden;
  margin: 10px 1%;
  min-width: 220px;
  max-width: 220px;
  max-height: 220px;
  min-height: 220px;
  width: 100%;
  text-align: center;
  box-shadow: 0 0 5px rgba(0,0,0,0.15);
}

.c0:hover img {
  opacity: 1;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0:hover h2 {
  -webkit-transform: skew(-10deg) rotate(-10deg) translate(-150%,-50%);
  -ms-transform: skew(-10deg) rotate(-10deg) translate(-150%,-50%);
  transform: skew(-10deg) rotate(-10deg) translate(-150%,-50%);
}

.c0:hover:before {
  -webkit-transform: rotate(110deg) translateY(-150%);
  -ms-transform: rotate(110deg) translateY(-150%);
  transform: rotate(110deg) translateY(-150%);
}

.c0::before {
  height: 120%;
  width: 130%;
  top: 0;
  left: 0;
  content: '';
  position: absolute;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transform: rotate(110deg) translateY(-33%) translateX(-7px);
  -ms-transform: rotate(110deg) translateY(-33%) translateX(-7px);
  transform: rotate(110deg) translateY(-33%) translateX(-7px);
}

.c0 * {
  box-sizing: border-box;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c0 a {
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  position: absolute;
  z-index: 1;
  padding: 0;
  border: none;
}

<figure
  className="c0"
/>
`;

exports[`Renders Name 1`] = `
.c0 {
  position: absolute;
  left: 20px;
  right: 20px;
  display: inline-block;
  -webkit-transform: skew(-10deg) rotate(-10deg) translate(0,-50%);
  -ms-transform: skew(-10deg) rotate(-10deg) translate(0,-50%);
  transform: skew(-10deg) rotate(-10deg) translate(0,-50%);
  padding: 12px 5px;
  margin: 0;
  bottom: 0;
  text-transform: uppercase;
  font-weight: 400;
}

.c0 span {
  font-weight: 800;
  display: block;
}

<h2
  className="c0"
/>
`;
